### PR TITLE
ames: Deduplicate long messages

### PR DESCRIPTION
**Pull request by philipcmonk**
This is essentially what was pushed to ~marzod and ~marnus over the weekend.

This reduces the memory usage of Ames in some situations by delaying splitting messages into fragments until the last moment.  It also checks whether new messages longer than 1KB exist in queues to other ships or flows.  This deduplicates large messages, for example OTA updates that may need to be sent to many ships.

### CODE REVIEW

File: pkg/arvo/sys/vane/ames.hoon [hunk 1]
```diff
@@ -3377,6 +3415,17 @@
   |=  [=symmetric-key plaintext=shut-packet]
   ^-  @
   ::
+  =.  meat.plaintext
```
Comment (belisarius222) [inline pkg/arvo/sys/vane/ames.hoon:3418]: could be a =?

File: pkg/arvo/sys/vane/ames.hoon [hunk 2]
```diff
@@ -2222,7 +2255,8 @@
       =/  =message-pump-state
         (~(gut by snd.peer-state) bone *message-pump-state)
       ::
-      =/  message-pump    (make-message-pump message-pump-state channel)
+      =/  message-pump
```
Comment (belisarius222) [inline pkg/arvo/sys/vane/ames.hoon:2258]: It doesn't look like we need to thread peers.ames-state into this core anymore.

Comment (philipcmonk) [reply pkg/arvo/sys/vane/ames.hoon:2258]: GOOD point

**Comment by philipcmonk** [comment]
@belisarius222 Can I get a re-review?

**Comment by philipcmonk** [comment]
Whoops, accidentally merged it anyway.  Uh, let me know if something's wrong

**Pull request merged**